Framework/DataHandling/inc/MantidDataHandling/SaveCanSAS1D.h

```diff
@@ -80,6 +80,7 @@ Required properties:
 */
 class DLLExport SaveCanSAS1D : public API::Algorithm {
 public:
+  virtual ~SaveCanSAS1D() = default;
```

Comment:
There are couple of places where the algorithms are not `final` but it doesn't look they are used as bases. Have I got that wrong here and this is used somewhere along with the others?

Reply:
In this case it's a weird dependency tree from the V2 alg:
https://github.com/mantidproject/mantid/blob/01916cddab80834251ab4fca9aa58389ae6c69f4/Framework/DataHandling/inc/MantidDataHandling/SaveCanSAS1D2.h#L63

I've re-applied final to all algorithms and manually removed the keyword from the ones which the compiler flags to be 100% sure though (there's only ~20-30 so it was faster to do by hand)